politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +395,246 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
 	return &sr, nil
 }
 
+// comments requests all comments on a record from the comments plugin.
+func (p *piPlugin) comments(token []byte) (*comments.GetAllReply, error) {
+	reply, err := p.backend.PluginRead(token, comments.PluginID,
+		comments.CmdGetAll, "")
+	if err != nil {
+		return nil, err
+	}
+	var gar comments.GetAllReply
+	err = json.Unmarshal([]byte(reply), &gar)
+	if err != nil {
+		return nil, err
+	}
+	return &gar, nil
+}
+
+// isAncestorOf returns whether the given parentID is an ancestor of the given
+// childID comment, by traveling the comment tree defined by
+// the ParentID of the child comment.
+func (p *piPlugin) isAncestorOf(ancestorID, childID uint32, cs []comments.Comment) bool {
+	if ancestorID == childID {
+		return true
+	}
+	// Convert comments slice to a map
+	commentsMap := make(map[uint32]comments.Comment, len(cs))
+	for _, c := range cs {
+		commentsMap[c.CommentID] = c
+	}
+	// Travel the comment tree to search for an ancestor
+	for {
+		current := commentsMap[childID]
+		// If we reach the tree haed without visiting the provided
+		// ancestorID then it is not an ancestor.
+		if current.ParentID == 0 {
+			break
+		}
+		// Check if next parent in the tree is the ancestor
+		if current.ParentID == ancestorID {
+			return true
+		}
+		childID = current.ParentID
+	}
+	return false
+}
+
+// latestAuthorUpdate gets the latest author update on a record, if
+// the record has no author update it returns nil.
+func (p *piPlugin) latestAuthorUpdate(token []byte, cs []comments.Comment) (*comments.Comment, error) {
```

Comment:
```suggestion
func latestAuthorUpdate(token []byte, cs []comments.Comment) (*comments.Comment, error) {
```

Same reason as `isAcenstorOf`.